### PR TITLE
Build mongoDB indices

### PR DESCRIPTION
**Pull request by shuplenkov**

### CODE REVIEW

File: lib/stateView/contract/createContractDatabaseFactory.js [hunk 1]
```diff
@@ -12,13 +14,20 @@ function createContractDatabaseFactory(createSVDocumentRepository) {
    * @returns {Promise<*[]>}
    */
   async function createContractDatabase(svContract) {
-    const documents = svContract.getDataContract().getDocuments();
+    const dataContract = svContract.getDataContract();
+    const documents = dataContract.getDocuments();
     const contractId = svContract.getId();
 
     const promises = Object.keys(documents).map((documentType) => {
```
Comment (shumkov) [inline lib/stateView/contract/createContractDatabaseFactory.js]: Pass convertToMongoDbIndices as a dependecy?

File: test/integration/stateView/document/mongoDbRepository/SVDocumentMongoDbRepository.spec.js [hunk 2]
```diff
@@ -684,6 +684,36 @@ describe('SVDocumentMongoDbRepository', function main() {
       expect(collectionsAfter).to.have.lengthOf(1);
       expect(collectionsAfter[0].collectionName).to.equal(svDocumentRepository.getCollectionName());
     });
+
```
Comment (shumkov) [inline test/integration/stateView/document/mongoDbRepository/SVDocumentMongoDbRepository.spec.js]: We need to test `createIndices` as well. In `createCollection` you can check that you just call `createIndices()`

File: lib/stateView/contract/createContractDatabaseFactory.js [hunk 3]
```diff
@@ -12,13 +13,20 @@ function createContractDatabaseFactory(createSVDocumentRepository) {
    * @returns {Promise<*[]>}
    */
   async function createContractDatabase(svContract) {
```
Comment (shumkov) [inline lib/stateView/contract/createContractDatabaseFactory.js]: I did not find unit test.

Comment (shuplenkov) [reply lib/stateView/contract/createContractDatabaseFactory.js]: Yes. I'll create it now.

File: test/integration/stateView/document/mongoDbRepository/SVDocumentMongoDbRepository.spec.js (hunk 2 above)
Comment (shuplenkov) [reply test/integration/stateView/document/mongoDbRepository/SVDocumentMongoDbRepository.spec.js]: We check this in the next test

File: test/unit/stateView/contract/createContractDatabaseFactory.spec.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+    convertToMongoDbIndices = this.sinon.stub();
+
+    createContractDatabase = createContractDatabaseFactory(
+      createSVDocumentRepository,
+      convertToMongoDbIndices,
+    );
+  });
+
+  it('should create all collections with indices', async () => {
```
[the remark this answers is not in the document]
Comment (shuplenkov) [reply test/unit/stateView/contract/createContractDatabaseFactory.spec.js]: createSVDocumentRepository is a dependency.

File: test/integration/stateView/document/mongoDbRepository/SVDocumentMongoDbRepository.spec.js (hunk 2 above)
Comment (shumkov) [reply test/integration/stateView/document/mongoDbRepository/SVDocumentMongoDbRepository.spec.js]: I don't see: `describe('#createIndices')` in this suite.

**Pull request merged**